src/webview/chat.js

```diff
@@ -129,20 +129,28 @@ const LINE_HEIGHT = 36;
 
   function addMessageToUI(result) {
     if (result.role === "assistant") {
-      addAssistantMessageToUI(result);
+      withScroll(addAssistantMessageToUI)(result);
     } else {
-      addUserMessageToUI(result.textContent);
-      addAssistantThinkingMessageToUI();
+      withScroll(addUserMessageToUI, false)(result.textContent);
+      withScroll(addAssistantThinkingMessageToUI, false)();
     }
   }
 
-  function withStickyScroll(wrapped) {
-    return function () {
-      const { scrollHeight: scrollHeightBefore } = messageContainer;
-      wrapped.apply(this, arguments);
-      const { scrollHeight: scrollHeightAfter } = messageContainer;
-      stickyScrollToBottom(scrollHeightAfter - scrollHeightBefore);
-    };
+  function withScroll(wrapped, stickyOnly = true) {
```

Comment:
it's not clear from this what stickyOnly means

Reply:
Extracted to a separate function - there wasn't much overlapping functionality so this is neater anyway.